checks.d/mysql.py

```diff
@@ -119,153 +381,272 @@ def _connect(self, host, port, mysql_sock, user, password, defaults_file):
+            try:
+                results['information_schema_size'] = self._query_size_per_schema(db)
+                metrics.update(SCHEMA_VARS)
+            except Exception as e:
+                self.log.debug("Schema size metrics unavailable at this time: {0}".format(e))
+
+        if _is_affirmative(options.get('replication', False)):
+            # Get replica stats
+            results.update(self._get_replica_stats(db))
+            metrics.update(REPLICA_VARS)
+
             # get slave running form global status page
-            slave_running = self._collect_string('Slave_running', status_results)
+            slave_running_status = AgentCheck.UNKNOWN
+            slave_running = self._collect_string('Slave_running', results)
             if slave_running is not None:
                 if slave_running.lower().strip() == 'on':
                     slave_running = 1
+                    slave_running_status = AgentCheck.OK
                 else:
                     slave_running = 0
-                self.gauge("mysql.replication.slave_running", slave_running, tags=tags)
-            self._collect_dict(
-                GAUGE,
-                {"Seconds_behind_master": "mysql.replication.seconds_behind_master"},
-                "SHOW SLAVE STATUS", db, tags=tags
-            )
+                    slave_running_status = AgentCheck.CRITICAL
+                # deprecated in favor of service_check("mysql.replication.slave_running")
+                self.gauge(self.SLAVE_SERVICE_CHECK_NAME, slave_running, tags=tags)
+            else:
+                # MySQL 5.7.x might not have 'Slave_running'. See: https://bugs.mysql.com/bug.php?id=78544
+                # look at replica vars collected at the top of if-block
+                ver = self._get_version(db, host)
+                if (ver[0], ver[1], ver[2]) >= VER_5_7:
+                    slave_io_running = self._collect_string('Slave_IO_Running', results)
+                    slave_sql_running = self._collect_string('Slave_SQL_Running', results)
+                    if slave_io_running:
+                        slave_io_running = (slave_io_running.lower.strip() == "yes")
+                    if slave_sql_running:
+                        slave_sql_running = (slave_sql_running.lower.strip() == "yes")
+
+                    if not (slave_io_running is None and slave_sql_running is None):
+                        if slave_io_running and slave_sql_running:
+                            slave_running_status = AgentCheck.OK
+                        elif not slave_io_running and not slave_sql_running:
+                            slave_running_status = AgentCheck.CRITICAL
+                        else:
+                            # not everything is running smoothly
+                            slave_running_status = AgentCheck.WARNING
+
+            self.service_check(self.SLAVE_SERVICE_CHECK_NAME, slave_running_status, tags=tags)
```

Comment:
Let's send this service check with the `service_check_tags` instead of the metrics' tags.